### PR TITLE
fix: error when serve --production using bento path

### PR DESCRIPTION
**Pull request by parano**
## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/server/__init__.py [hunk 1]
```diff
@@ -116,19 +118,19 @@ def serve_production(
...
         )
+        sockets_map[runner_name] = runner_socket
         cmd_runner = f"""import bentoml._internal.server
 bentoml._internal.server.start_prod_runner_server(
     "{bento_identifier}",
     "{runner_name}",
     working_dir="{working_dir}",
     instance_id=$(CIRCUS.WID),
-    fd=$(circus.sockets.{runner_name}),
+    fd={runner_socket.fileno()},
```
Comment (parano) [inline bentoml/_internal/server/__init__.py:133]: @bojiang I'm not sure why circus is not replacing it here, but changing it to `.fileno` call seems to be working well

File: bentoml/_internal/server/__init__.py [hunk 2]
```diff
@@ -159,6 +163,7 @@ def serve_production(
             numprocesses=app_workers or 1,
             stop_children=True,
             use_sockets=True,
+            working_dir=working_dir,
```
Comment (parano) [inline bentoml/_internal/server/__init__.py:166]: @bojiang we need this change per https://github.com/bentoml/BentoML/pull/2064#discussion_r759091488

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -6,10 +6,10 @@
 from bentoml._internal.models import ModelStore
 
 if TYPE_CHECKING:
-    from _pytest.config import Config
-    from _pytest.config.argparsing import Parser
     from _pytest.nodes import Item
+    from _pytest.config import Config
```
Comment (parano) [inline tests/integration/conftest.py:10]: auto formatting change caused by another PR

File: bentoml/_internal/server/__init__.py (hunk 1 above)
Comment (bojiang) [reply bentoml/_internal/server/__init__.py:133]: because we didn't ensure the runner_name to be a valid identifier.
Fixed in #2106

**Pull request merged**